### PR TITLE
Translate plain dispenser module to Elixir

### PR DESCRIPTION
**Pull request by frojasg**

### CODE REVIEW

File: dispenser_plain_elixir/dispenser/server.ex [hunk 1]
```diff
@@ -0,0 +1,45 @@
+defmodule Dispenser.Server do
```
Comment (minostro) [inline dispenser_plain_elixir/dispenser/server.ex:1]: is it possible to do something like this?

``` elixir
defmodule Dispenser do
  defmodule Server do
  end
end
```

File: dispenser_plain_elixir/dispenser/server.ex [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      {:reset} ->
+        loop(1)
+      {:stop} ->
+        true
+    end
+  end
+
+  # # API
+  def start() do
+    spawn(Dispenser.Server, :loop, [1])
```
Comment (minostro) [inline dispenser_plain_elixir/dispenser/server.ex:19]: is it possible to user just `Server` here?

File: dispenser_plain_elixir/dispenser/server.ex [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  end
+
+  def take_ticket(pid) do
+    ref = make_ref()
+    send pid, {self(), ref, :take_ticket}
+    receive do
+      {^pid, ^ref, msg} -> msg
```
Comment (minostro) [inline dispenser_plain_elixir/dispenser/server.ex:26]: Does `^` I want the value of the "thing" on pattern matching?

**Comment by minostro** [comment]
this is very cool

File: dispenser_plain_elixir/dispenser/server.ex (hunk 1 above)
Comment (frojasg) [inline dispenser_plain_elixir/dispenser/server.ex:1]: yes, nesting module is possible.

File: dispenser_plain_elixir/dispenser/server.ex (hunk 2 above)
Comment (frojasg) [inline dispenser_plain_elixir/dispenser/server.ex:19]: it's possible but we would have to change the defmodule definition to 

``` elixir
 defmodule Dispenser do
  defmodule Server do
    ...
  end
end
```

File: dispenser_plain_elixir/dispenser/server.ex (hunk 3 above)
Comment (frojasg) [inline dispenser_plain_elixir/dispenser/server.ex:26]: Yes, elixir let you rebind variable names, that's why you need to explicitly tell elixir to use that value.

**Pull request merged**